### ISSUE TITLE
Computer labs Data

### ISSUE THREAD

**Issue by vonbearshark** [Pitt-CSC/club_project_2017#35]
Pitt has an API for the computer labs status--we should use it! This is likely a one-off call for the Pitt data, and not a part of the WPRDC function call system (#17). Let's just get the call set up and go from there.

**Comment by SPN11** [comment]
@vonbearshark I'd be interested in taking this. As for the API, where can I find it? I saw @RitwikGupta wrote a PittAPI a while back, but this'll be part of main.js right? And in terms of user interaction, would the labs appear on the map as pins and if you click on them they'll appear with information? Or will there be a small section on the sidebar that has the different computing labs and their availability?

**Comment by vonbearshark** [comment]
All good questions. Here's the situation: we will be using that PittAPI you mentioned. The club has absorbed the responsibilities of maintaining and extending Ritwik's API (found here: https://github.com/Pitt-CSC/PittAPI). Our plan is to get a server on the CS Dept's network and host the API as a Flask application, using this wrapper repo(that the club now also maintains): https://github.com/Pitt-CSC/PittAPIWebWrapper. We'd call the API from within the JS and then iterate on the data that is returned. However, we're still getting the confirmation for that, so we don't have this hosted at all right now.

Right now, all the calls from `main.js` are coming from the WPRDC datasets. This will be the first call to the PittAPI. So we don't have boilerplate for how we'll be interacting with it necessarily. Still, take a look at how we use the `fetchWPRDCData` function, because it will be as similar to that as possible. My suggestion would be to host the API locally and hit `localhost` instead of the remote endpoint, then we can swap in the proper URL when it's active. @RitwikGupta should be able to help you set that up and provide a little documentation with how the computer labs endpoint works, exactly.

I'm open to whatever UI suggestions you come up with, but I was thinking that, yeah, there would be a pin at every computer lab site this supports, and when you click on it, the popup would show the availability and maybe a color that designated their status.

Right now this is a little nebulous, and we're sorry about that. But we'll be happy to work with you as closely as possible. Getting this set up would be really big for the project.

After you accept the invitation to collaborate, I can assign you to this.

**Comment by vonbearshark** [comment]
If you can even just get the data flow for now, we have a separate Issue open for the UI representation #36 . So you can move on to that after this, or tackle something else.

**Comment by SPN11** [comment]
Ok thanks so much! I'll study `fetchWPRDCData`, and I'm new to Flask and fairly new to Javascript but I'll do my best. I'll also talk to @RitwikGupta for how to go about hosting it locally and how to use his API. I was also thinking the pins would be most appropriate, but that'll be for #36

**Comment by SPN11** [comment]
@vonbearshark @RitwikGupta I've adapted to the Fetch API and now I'm getting nicely-structured JSON responses for each computing lab.

**Comment by vonbearshark** [comment]
We still want this to be a marker, so we'll need to find a way to create a pin like we do with the others. The label should be a computer icon. The hardest part will probably be mapping the computer labs to lat and long locations. We'll have to hard-code the names of the labs to their locations (which you'll have to look up). You don't have to worry about the pin popup, we'll handle that in #36. Look how the Library data is mapped, because it is similar in that it doesn't have a date associated

**Comment by SPN11** [comment]
Oh yeah of course! I assumed we would have to hard-code the lat and long coords. All of that would be related to #36 though, right? Do we have a computer icon yet?

**Comment by RitwikGupta** [comment]
Specifically the locations are:

```python
location_dict = {
       'ALUMNI': 0,
       'BENEDUM': 1,
       'CATH_G26': 2,
       'CATH_G27': 3,
       'LAWRENCE': 4,
       'HILLMAN': 5,
       'SUTH': 6
}
```

**Comment by SPN11** [comment]
I've got data for each of those labs. I can look them up on a map now.

**Comment by vonbearshark** [comment]
> Oh yeah of course! I assumed we would have to hard-code the lat and long coords. All of that would be related to #36 though, right? Do we have a computer icon yet?

@SPN11 We should at least add the pin and icon to the map in this one because that involves mapping the name of the lab with a hard-coded lat long location, and that's still data processing, I think. All of our icons come from fontawesome right now, so this one works: http://fontawesome.io/icon/desktop/. Take a look at how we create the icons with the WPRDC data: 
- Icon data stuff: https://github.com/Pitt-CSC/PantherView/blob/master/main.js#L148
- mapped to dataset: https://github.com/Pitt-CSC/PantherView/blob/master/main.js#L191
- added as part of the map: https://github.com/Pitt-CSC/PantherView/blob/master/main.js#L352
(the `title` for this should be the name of the lab)

Since each of these Pitt API calls are unique we don't need to have your function be that general; you can add the icon right inside of the function where you're working with the `data` JSON response itself.

Essentially I think you should have a `getPittComputerLabData` function that takes a location (name), location (lat, long) and icon (at least) and creates a pin and adds it to the `markers` array, then call that with all the locations and designated pins.

> Specifically the locations are:

@RitwikGupta I don't think the Flask wrapper accommodates that, we just pass the name of the location, I think and the PittAPI lib handles that indexing system. I'm talking about the actual lat long location

**Comment by SPN11** [comment]
For my own reference:
**Alumni**
Lat: 40.445763°
Long: -79.953834°

**Benedum**
Lat: 40.443844°
Long: -79.958475°

**Cath_G26**
Lat: 40.444209°
Long: -79.95309°

**Cath_G27**
Lat: 40.444326°
Long: -79.953286°

**Lawrence**
Lat: 40.442277°
Long: -79.955023°

**Hillman**
Lat: 40.442787°
Long: -79.953942°

**Sutherland**
Lat: 40.445953°
Long: -79.962444°

And ok! I'll take a look at those references.

**Comment by vonbearshark** [comment]
Might have to move Cathy_G26 and G27 off by one or something because they're the exact same. Moving one or two should keep it within Cathy but slightly removed when you zoom way in

**Comment by SPN11** [comment]
True. I'll tweak the values a bit.

**Comment by SPN11** [comment]
Should I put computer labs under the same dataset as the others (WPRDC_DATA_SOURCES) or should I make a separate one for Pitt-related sources?

**Comment by vonbearshark** [comment]
Good question. I think you could even have a separate one just for the Pitt computer labs, and we'll create another for the Pitt laundry, etc, because again these Pitt API calls are pretty unique and not as standardized. Each one is a different kind of data

**Comment by SPN11** [comment]
All good thoughts. I'll make a totally separate one for Pitt labs then.

**Comment by vonbearshark** [comment]
Closed by #129
